### ISSUE TITLE
Why Isn't watchOS a Supported Destination?

### ISSUE THREAD

**Issue by VaslD** [yonaskolb/XcodeGen#1441]
According to the Project Spec, valid [`platform`](https://github.com/yonaskolb/XcodeGen/blob/master/Docs/ProjectSpec.md#platform)s are:

- `iOS`
- `tvOS`
- `macOS`
- `watchOS`
- `visionOS`

But valid [`supportedDestinations`](https://github.com/yonaskolb/XcodeGen/blob/master/Docs/ProjectSpec.md#supported-destinations) are:

- `iOS`
- `tvOS`
- `macOS`
- `macCatalyst`
- `visionOS`

Which means the following project definition generates a multiplatform (Xcode 14+) `ArgumentParser.framework` available to all Apple platforms except watchOS:

```yaml
targets:
  ArgumentParser:
    type: framework
    # platform: auto (implicit)
    supportedDestinations:
      - iOS
      - macOS
      - tvOS
      - watchOS # unrecognized
      - visionOS
    sources:
      - path: "ArgumentParser/Sources/ArgumentParser"
        includes:
          - "**/*.swift"
    settings:
      BUILD_LIBRARY_FOR_DISTRIBUTION: YES
      SKIP_INSTALL: NO
      MAKETING_VERSION: "1.3.0"
```

I don't understand why `watchOS` was left unsupported. If I first generated a project without watchOS support, and then added watchOS destination in the Xcode project editor, the only changes appeared on Git were:

```diff
-                SUPPORTED_PLATFORMS = "appletvos appletvsimulator iphoneos iphonesimulator macosx xros xrsimulator";
+                SUPPORTED_PLATFORMS = "appletvos appletvsimulator iphoneos iphonesimulator macosx watchos watchsimulator xros xrsimulator";
-                TARGETED_DEVICE_FAMILY = "1,2,3,7";
+                TARGETED_DEVICE_FAMILY = "1,2,3,4,7";
```

It didn't seem like there would be a technically restriction, or even too much work, on adding watchOS as a `supportedDestinations`.

**Comment by giginet** [comment]
This will be resolved by https://github.com/yonaskolb/XcodeGen/pull/1438

**Comment by zewuchen** [comment]
This issue was fixed in [Release 2.41.0](https://github.com/yonaskolb/XcodeGen/commit/6bbf2c6543593a0bafa240abdcb045cd04736661). Can you close?
